src/StackExchange.Utils.Http/DefaultHttpClientPool.cs

```diff
@@ -62,11 +69,45 @@ private HttpClient CreateHttpClient(HttpClientCacheKey options)
         /// </summary>
         public void Clear() => ClientPool.Clear();
 
-        private struct HttpClientCacheKey
+        private struct HttpClientCacheKey : IEquatable<HttpClientCacheKey>
         {
             public TimeSpan Timeout { get; }
+            public IWebProxy Proxy { get; }
+
+            public HttpClientCacheKey(TimeSpan timeout, IWebProxy proxy)
+            {
+                Timeout = timeout;
+                Proxy = proxy;
+            }
+
+            public override bool Equals(object obj)
+            {
+                return obj is HttpClientCacheKey key && Equals(key);
+            }
```

Comment:
Can we please move the 1-liners to `=>` just to be concise? (happy to do this as follow-up to merge too)